### DISCUSSION TITLE
Difference between Github.io and local browser?

### DISCUSSION THREAD

**Discussion by TehBrian** [community/community#22447] [Other Feature Feedback, Questions, & Ideas]
<p>If you go to my GitHub Pages site at <a href="https://t3hbrian.github.io" rel="nofollow noopener">https://t3hbrian.github.io</a>, you can see that all the text is quite big. </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/9d32a34f068f92ed668f60eebb265fd1274aa716.png" data-download-href="/uploads/short-url/mqDs9BYGd9HGRffBApzEQntpKlw.png?dl=1" title="Screen Shot 2018-10-17 at 12.27.18 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/32250137/181070961-ab2ae9a1-cff4-4213-b98f-f78cc6b03147.png" alt="Screen Shot 2018-10-17 at 12.27.18 AM.png" data-base62-sha1="mqDs9BYGd9HGRffBApzEQntpKlw" width="690" height="431" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/9d32a34f068f92ed668f60eebb265fd1274aa716_2_690x431.png, https://aws1.discourse-cdn.com/github/optimized/1X/9d32a34f068f92ed668f60eebb265fd1274aa716_2_1035x646.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/9d32a34f068f92ed668f60eebb265fd1274aa716.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/9d32a34f068f92ed668f60eebb265fd1274aa716_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2018-10-17 at 12.27.18 AM.png</span><span class="informations">1280×800 82.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>I’m on Mac, so this might differ a bit, but if you go to your terminal and you type “git clone <a href="https://github.com/t3hbrian/t3hbrian.github.io" rel="nofollow noopener">https://github.com/t3hbrian/t3hbrian.github.io</a>” and then you open the index.html file locally you can see that the font size is much smaller, and much more natural looking.
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/c82b4139f6f8acded339e41ed39ac20b7ee5bf55.png" data-download-href="/uploads/short-url/syMam93vUZjG6JpUf7tfjAgi4Bv.png?dl=1" title="Screen Shot 2018-10-17 at 12.27.53 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/32250137/181070962-2b5a5c1d-7822-4032-af7b-f0457ef9c71c.png" alt="Screen Shot 2018-10-17 at 12.27.53 AM.png" data-base62-sha1="syMam93vUZjG6JpUf7tfjAgi4Bv" width="690" height="431" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/c82b4139f6f8acded339e41ed39ac20b7ee5bf55_2_690x431.png, https://aws1.discourse-cdn.com/github/optimized/1X/c82b4139f6f8acded339e41ed39ac20b7ee5bf55_2_1035x646.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/c82b4139f6f8acded339e41ed39ac20b7ee5bf55.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/c82b4139f6f8acded339e41ed39ac20b7ee5bf55_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2018-10-17 at 12.27.53 AM.png</span><span class="informations">1280×800 81.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>It’s still the same font, Roboto, so it is working correctly. I’m using REMs which refers to the html root element for the size to set the font.
<p>You can take a peek at the CSS in the resources folder, but I set the html root element to 62.5% of the regular font size, so my best guess was that Github Pages is messing with the number and putting the number at 100%.</p>
<p>However, I put that theory to the test and I changed the html root element font size locally to 100% and it still was a different size. So I’m pretty stuck and this is super annoying because it’s the exact same thing but it’s loading differently.</p>
<p>I’m not really sure what to do and I couldn’t find anyone with a similar problem, since I’m only using plain html and css. Thanks!</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/itstehbrian">@itstehbrian</a>,</p>
<p>Thank you for being here! I think this might have to do with Roboto itself, not the way Pages is rendering, could you try and test a different font?</p>

**Comment by TehBrian** [comment]
<p>Hi <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a>!</p>
<p>Thank you for responding. Sorry I posted twice, there was a bug for some reason the first post got deleted and then put back and I couldn’t delete it after that.</p>
<p>I changed the font to Helvetica which you can see by the console on both the images and the ‘e’, as Roboto has a much more relaxed ‘e’ and Helvetica has the tip of the ‘e’ perfectly straight.</p>
<p>Here is the image for the site hosted at Pages. </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/bb386085ded4692405c40d8c2a20fbfda95cd64f.png" data-download-href="/uploads/short-url/qIe4BliDgGvR7ISCmzVEeVme6ET.png?dl=1" title="Screen Shot 2018-10-20 at 12.40.59 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/32250137/181070965-52a29190-4b89-4757-9f54-272cf554129f.png" alt="Screen Shot 2018-10-20 at 12.40.59 AM.png" data-base62-sha1="qIe4BliDgGvR7ISCmzVEeVme6ET" width="690" height="431" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/bb386085ded4692405c40d8c2a20fbfda95cd64f_2_690x431.png, https://aws1.discourse-cdn.com/github/optimized/2X/b/bb386085ded4692405c40d8c2a20fbfda95cd64f_2_1035x646.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/b/bb386085ded4692405c40d8c2a20fbfda95cd64f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/bb386085ded4692405c40d8c2a20fbfda95cd64f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2018-10-20 at 12.40.59 AM.png</span><span class="informations">1280×800 106 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>And here is the same exact site, copied by putting ‘git clone <a href="https://github.com/t3hbrian/t3hbrian.github.io" rel="nofollow noopener">https://github.com/t3hbrian/t3hbrian.github.io</a>’ and then making the change to Helvetica, then pushing *all* of the changes and files into the master at GitHub. </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/39250576e0aa7a2c6c14399ab1367b7d7fb7aee1.png" data-download-href="/uploads/short-url/89wx5tk2Bh52ytzXE3EQKib8TdL.png?dl=1" title="Screen Shot 2018-10-20 at 12.41.02 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/32250137/181070966-89fe52ad-17fa-41fd-958c-aca7473fc18f.png" alt="Screen Shot 2018-10-20 at 12.41.02 AM.png" data-base62-sha1="89wx5tk2Bh52ytzXE3EQKib8TdL" width="690" height="431" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/39250576e0aa7a2c6c14399ab1367b7d7fb7aee1_2_690x431.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/39250576e0aa7a2c6c14399ab1367b7d7fb7aee1_2_1035x646.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/3/39250576e0aa7a2c6c14399ab1367b7d7fb7aee1.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/39250576e0aa7a2c6c14399ab1367b7d7fb7aee1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2018-10-20 at 12.41.02 AM.png</span><span class="informations">1280×800 107 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>As you can see, the one hosted at GitHub pages has significantly larger text for no reason. As you can see the font size is set in REMs. For anyone else reading this who aren’t familiar with those, they are a size BASED on the root html element font size, which is where it gets its ‘r’ in the title.</p>
<p>You can see in the screenshot that the root is 62.5% of the default font size. This is for those who change their default font size that the website would adapt to that font. The default for everyone who does not change it, however, is 100% = 16px. So if you go into the REM and you type in 1 REM it would put it as 16px. To make the math easier, though, you can set it to 62.5%, which makes 1 REM be 10px, 0.1 REM be 1px, you get the idea.</p>
<p>My thought is that perhaps GitHub Pages is enforcing a certain size, say, 14 pixels, overwriting the 62.5% size. I put this in the original post but I forgot that I just set it to 100%. GitHub would still, in this idea, overwrite that. But I tried it with both being set to 16 pixels.</p>
<p>Same deal, pushing the exact same thing. Here’s GitHub Pages.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/829e199328fa648c1ae19f621cf3ed74e3f181d5.png" data-download-href="/uploads/short-url/iDuNvXeGvTXHqQ3YUktBEUHhMXz.png?dl=1" title="Screen Shot 2018-10-20 at 12.50.54 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/32250137/181070967-b8a62e28-308a-4e2e-8f55-58d4717fe6b2.png" alt="Screen Shot 2018-10-20 at 12.50.54 AM.png" data-base62-sha1="iDuNvXeGvTXHqQ3YUktBEUHhMXz" width="690" height="431" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/829e199328fa648c1ae19f621cf3ed74e3f181d5_2_690x431.png, https://aws1.discourse-cdn.com/github/optimized/2X/8/829e199328fa648c1ae19f621cf3ed74e3f181d5_2_1035x646.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/8/829e199328fa648c1ae19f621cf3ed74e3f181d5.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/829e199328fa648c1ae19f621cf3ed74e3f181d5_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2018-10-20 at 12.50.54 AM.png</span><span class="informations">1280×800 107 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p> Here’s the exact same repository, exact same everything, opened on my computer locally.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d0960437acfad0d1e052fb0d4ee719b72c3ce7c4.png" data-download-href="/uploads/short-url/tLeINVY6rXnHV13Big0xGRnc0dK.png?dl=1" title="Screen Shot 2018-10-20 at 12.50.57 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/32250137/181070968-3b8efae1-1e0b-43dc-984a-6577fda35be9.png" alt="Screen Shot 2018-10-20 at 12.50.57 AM.png" data-base62-sha1="tLeINVY6rXnHV13Big0xGRnc0dK" width="690" height="431" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/d0960437acfad0d1e052fb0d4ee719b72c3ce7c4_2_690x431.png, https://aws1.discourse-cdn.com/github/optimized/2X/d/d0960437acfad0d1e052fb0d4ee719b72c3ce7c4_2_1035x646.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/d/d0960437acfad0d1e052fb0d4ee719b72c3ce7c4.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d0960437acfad0d1e052fb0d4ee719b72c3ce7c4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2018-10-20 at 12.50.57 AM.png</span><span class="informations">1280×800 105 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I’m really confused. I have no idea what’s going on and it should theoretically be the same, I’m pretty stuck and I can’t think of anything else.</p>
<p>Thanks, Brian.</p>

**Comment by sflanders95** [comment]
<p>I was able to reproduce exactly what is being displayed on <a href="https://t3hbrian.github.io/" rel="nofollow noopener">https://t3hbrian.github.io</a> locally.  I did the following:</p>
<pre><code>$ git clone https://github.com/t3hbrian/t3hbrian.github.io
$ cd t3hbrian.github.io/
$ jekyll serve --port 8001
</code></pre>
<p>So then I looked at your stylesheet.  You have huge values for the rem settings.  Remember that 8rem is 800% of the base html font value.  So I changed all of your 8rems to 1.8 rem which is what I think you meant.  so 4rem similarly became 1.4rem.  After that modifying main.css, the page looked normal for me.</p>
<pre><code>$ git diff resources/css/main.css
diff --git a/resources/css/main.css b/resources/css/main.css
index 8da24f1..96bd795 100644
--- a/resources/css/main.css
+++ b/resources/css/main.css
@@ -5,42 +5,42 @@ html {
 }

 body {
- padding: 0rem 5rem;
+ padding: 0rem 1.5rem;
        background-color: #faf8f8;
 }

 h1 {
- font: 400 8rem 'Roboto', sans-serif;
+ font: 400 1.8rem 'Roboto', sans-serif;
        color: #323030;
 }
/ * etc... */
</code></pre>

**Comment by TehBrian** [comment]
<p>Hi <a class="mention" href="/sflanders95">@sflanders95</a>.</p>
<p>First off I would just like to say I’m very inexperienced with Jekyll and I haven’t use it at all. And it’s not used in this project, so I’m not entirely sure how you’re using Jekyll in your terminal. Again I’m very inexperienced so I have no idea how it works.</p>
<p>Secondly, the size isn’t the point. I don’t mind <em>exactly</em> how big or small it is, l just previously believed that opening my website via double-clicking index.html would provide the same results as opening it on GitHub pages, since it’s serving the same content.</p>
<p>Adding to this, no matter how gigantic 8rem is, if you look at my previous post’s screenshots you can see that the font sizes for both websites are the <em>exact</em> same, you can see this because of the console in the bottom literally saying that it’s giving my browser the same font sizes. It’s not to do with what is actually being given to the web browser to load, since it’s the exact same, it’s <em>how</em> it’s loading it.</p>
<p>So, I did a bit of looking into of Jekyll it seems that all it is, is that you give it Markdown or any markup language and through some themes it makes it into a static website. Since what I’m doing is more of a venture into learning HTML, CSS, and JavaScript, I don’t want to start learning frameworks or whatever Jekyll is.</p>
<p>The entire problem is that I’m confused as to how either getting the file and all associated files through my computer, and opening it through double-clicking index.html is different than the <em>same file</em> and _same associated files _through GitHub Pages.</p>
<p>I also opened the website up through Google Chrome and it’s the same result as me opening it in Safari. It’s not how big or small the text size is itself, it’s just that it’s different depending on how you get the _same exact _file.</p>
<p>Is there something to do with me opening it through index.html? Should I start using something like Jekyll so I can go to localhost:8001? Am I doing something wrong?</p>
<p>Thanks, Brian</p>

**Comment by sflanders95** [comment]
<p><a class="mention" href="/itstehbrian">@itstehbrian</a>, since your site isn’t utilizing layouts you shouldn’t need jekyll.  I did that to most closely mimick the github pages server.  So I tried it again w/o jekyll:</p>
<pre><code>SF:git sflanders95$ git clone https://github.com/t3hbrian/t3hbrian.github.io
Cloning into 't3hbrian.github.io'...
remote: Enumerating objects: 59, done.
remote: Counting objects: 100% (59/59), done.
remote: Compressing objects: 100% (43/43), done.
remote: Total 369 (delta 27), reused 29 (delta 10), pack-reused 310
Receiving objects: 100% (369/369), 34.26 MiB | 8.97 MiB/s, done.
Resolving deltas: 100% (172/172), done.
SF:git sflanders95$ cd t3hbrian.github.io/
SF:t3hbrian.github.io sflanders95$ open index.html
</code></pre>
<p>The above session opened up chrome on my mac.  It exactly resembles the page on github down to the animated text.  My <a>chrome://version/</a> is 69.0.3497.100 (Official Build) (64-bit).  Having tried it two different ways and getting the same results my next suggestion would be to:</p>
<ol>
<li>use the “open index.html” statement in a terminal as I did above to ensure the correct file is being displayed. and…</li>
<li>hold the “Shift” key as you click on the chrome refresh button when viewing your local copy to ecourage chrome not to use any cache.  </li>
</ol>
<p><img src="https://user-images.githubusercontent.com/12764874/181070971-bb54197e-64c3-4fc8-beb9-2ff6e80076b9.jpeg" alt="Screenshot at Oct 20 19-11-08.jpg" data-base62-sha1="vWCzAwYr4taNNi0GSKQSzmaREfb" width="670" height="432"></p>
<p>Another thing that I noticed in local and on github pages is that when I “inspect” the page, then click on the html tag in the “elements” window, and look at the corresponding “styles” window, I see a yellow triangle next to the font-size: 62.5; item.   When I hover over the warning icon, it says invalid property value.  Do you get that in both online and local inspections.  Hope that helps.</p>
<p>-Steve</p>

**Comment by TehBrian** [comment, marked as answer]
<p>Hello! Thanks for the reply <a class="mention" href="/sflanders95">@sflanders95</a>. Also tagging <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> just so they can see the fix.</p>
<p>I’ll look into the issue with font-size: 62.5% showing up as an error, but onto more pressing matters.</p>
<p>I’ve been having this problem for a while, even for websites that were completely different but were on the <em>same repository</em>, and <em>named the same</em>. This problem has been happening for months. I think you know where I might be going with this.</p>
<p>My browser I usually use is Safari. You were testing on Google Chrome and I figured I should try that too, since maybe it’s a Safari only problem. I go to Google Chrome and the websites are the exact same. Same font size.</p>
<p>I start figuring Safari is bad for web development, but then I figure maybe Safari has it cached, just like you told me to reload the page while ignoring caches on Google Chrome. So I went into Safari and removed all caches for <a href="http://t3hbrian.github.io" rel="nofollow noopener">t3hbrian.github.io</a>. Still the problem remains. Then I remember…</p>
<p>On Safari you can use CMD plus the ‘+’ or ‘-’ keys to make the page bigger and smaller respectively. You can also use CMD plus ‘0’ to reset it. I figure, hey, maybe this entire problem was that Safari keeps the size you set for the website indefinitely, and apart from the cache/cookies, and I had made the page bigger before, and since I kept my repository for my website on my desktop folder all the time and never move it so the location would be the same…</p>
<p>I pressed CMD+0.</p>
<p>It became the same size as on GitHub Pages.</p>
<p>It seems that I was just receiving the error code 1D10T.</p>
<p>Thanks, this thread can be closed.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hey, <a class="mention" href="/sflanders95">@sflanders95</a> Thanks a lot for all your help here and <a class="mention" href="/itstehbrian">@itstehbrian</a> I’m glad you figured it out! Thank you for sharing the fix. We’re looking forward to seeing you around!</p>

**Comment by furkanmutlu** [comment]
<p>God! Thanks a lot for solution !</p>

**Comment by kingdomb** [comment]
<p>Wow! That really made me feel like an idiot. I was the one that set the default font size. OMG</p>

**Comment by Smith-Steve** [comment]
<p>I am having the same exact issue.</p>

**Comment by ansivakumar** [comment]
<p>I am having the same issue too, did you figure this out?</p>

**Comment by TehBrian** [comment]
<p>Yup. You can see my reply up above, but in a nutshell, my browser scaling was set higher for the local url.</p>

**Comment by davidicabello** [comment]
<p>Same happened to me! Holy molly , thanks for sharing the solution. Now i have to rezize everything lol.</p>
